### PR TITLE
Add test sets to tests

### PR DESCRIPTION
**Pull request by ssonal**
This pull-request adds test sets ( #18801 ) to:
- tuple tests ( #18807 )
- abstractarray tests ( #18809 )
- mpfr tests ( #18808 )

Most tests are covered, but there were some in abstractarray where I wasn't sure if I was doing things right. Happy to revisit and add those if these changes are alright. :smile:

### CODE REVIEW

File: test/abstractarray.jl [hunk 1]
```diff
@@ -340,91 +356,109 @@ function test_scalar_indexing{T}(::Type{T}, shape, ::Type{TestAbstractArray})
...
+            @test B[mask1, 1] == A[mask1, 1] == find(mask1)
+        end
+    end
 end
 
 function test_primitives{T}(::Type{T}, shape, ::Type{TestAbstractArray})
-    N = prod(shape)
-    A = reshape(collect(1:N), shape)
-    B = T(A)
+    @testset "test_primatives" begin
```
Comment (tkelman) [inline test/abstractarray.jl]: typo here. for these `test_...` functions I'd add the testset at the call site, not inside

Comment (ssonal) [inline test/abstractarray.jl]: This is the part I was uncertain about. Should there be one testset for each of the `test_...` function calls?

Comment (tkelman) [inline test/abstractarray.jl]: Probably, depends on how exactly they're called (I didn't go digging). You'd likely want the testset to show the input types, if those happen in a loop for example.

File: test/abstractarray.jl [hunk 2]
```diff
@@ -340,91 +356,109 @@ function test_scalar_indexing{T}(::Type{T}, shape, ::Type{TestAbstractArray})
...
+        B = T(A)
+        idxs = rand(1:N, 3, 3, 3)
+        @test B[idxs] == A[idxs] == idxs
+        @test B[vec(idxs)] == A[vec(idxs)] == vec(idxs)
+        @test B[:] == A[:] == collect(1:N)
+        @test B[1:end] == A[1:end] == collect(1:N)
+        @test B[:,:] == A[:,:] == reshape(1:N, shape[1], prod(shape[2:end]))
+        @test B[1:end,1:end] == A[1:end,1:end] == reshape(1:N, shape[1], prod(shape[2:end]))
+
+        @testset "Test with containers that aren't Int[]" begin
```
Comment (ssonal) [inline test/abstractarray.jl]: @tkelman Maybe I should also avoid using testsets for comments inside functions like I've done here?

File: test/abstractarray.jl [hunk 3]
```diff
@@ -674,22 +692,22 @@ function test_UInt_indexing(::Type{TestAbstractArray})
     end
 end
 
-# Issue 13315
 function test_13315(::Type{TestAbstractArray})
+    # Issue 13315"
```
Comment (tkelman) [inline test/abstractarray.jl]: quotes aren't really needed here

File: test/mpfr.jl [hunk 4]
```diff
@@ -826,50 +889,57 @@ tol = 1e-3
...
+@testset "issue #8318" begin
+    @test convert(Int64,big(500_000_000_000_000.)) == 500_000_000_000_000
+end
 
-# issue #9816
-# check exponent range is set to max possible
-@test MPFR.get_emin() == MPFR.get_emin_min()
-@test MPFR.get_emax() == MPFR.get_emax_max()
+@testset "issue #9816" begin
+    @testset "check exponent range is set to max possible" begin
```
Comment (tkelman) [inline test/mpfr.jl]: no need for the nesting here - just combine the comments into the same label?

File: test/tuple.jl [hunk 5]
```diff
@@ -41,127 +41,134 @@ end
...
+
+        @testset "1 argument" begin
+            @test map(foo, ()) === ()
+            @test map(foo, (1,)) === (2,)
+            @test map(foo, (1,2)) === (2,4)
+            @test map(foo, (1,2,3,4)) === (2,4,6,8)
+            @test map(foo, longtuple) === ntuple(i->2i,20)
+        end
+        @testset "2 argument" begin
+        @test map(foo, (), ()) === ()
```
Comment (tkelman) [inline test/tuple.jl]: wrong indent, "2 arguments" should be plural

File: test/tuple.jl [hunk 6]
```diff
@@ -86,12 +86,13 @@ begin
...
         end
-        @testset "2 argument" begin
-        @test map(foo, (), ()) === ()
-        @test map(foo, (1,), (1,)) === (2,)
-        @test map(foo, (1,2), (1,2)) === (2,4)
-        @test map(foo, (1,2,3,4), (1,2,3,4)) === (2,4,6,8)
-        @test map(foo, longtuple, longtuple) === ntuple(i->2i,20)
+
+        @testset "2 arguments" begin
+          @test map(foo, (), ()) === ()
```
Comment (tkelman) [inline test/tuple.jl]: multiples of 4 spaces, see CONTRIBUTING.md

**Comment by pkofod** [comment]
Comments seem addressed the same day this was reviewed. Can the conflicts be sorted out using Github's new functionality if it is ready?

**Comment by StefanKarpinski** [comment]
I'm sorry we let this slip through the cracks. I took a look but the conflict areas are pretty large so it will require a bit more work than I've got time for right now.

**Comment by fredrikekre** [comment]
#22606, #20471 and #23863. For future reference it is easier to review smaller PR's.

**Pull request closed without merging**